packages/react/src/VideoPlayer/VideoPlayer.tsx

```diff
@@ -0,0 +1,213 @@
+import React, {useState, useRef, useEffect, useCallback} from 'react'
+import clsx from 'clsx'
+import type {BaseProps} from '../component-helpers'
+import {Text} from '../Text'
+import {Controls} from './components'
+import {MarkGithubIcon} from '@primer/octicons-react'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/video-player/base.css'
+
+/** * Main Stylesheet (as a CSS Module) */
+import styles from './VideoPlayer.module.css'
+
+type VideoPlayerProps = {
+  poster?: string
+  title: string
+  branding?: boolean
+  children: React.ReactElement | React.ReactElement[]
+  ref?: React.RefObject<HTMLVideoElement>
+} & React.HTMLProps<HTMLVideoElement> &
+  BaseProps<HTMLVideoElement>
+
+const Root = ({
+  poster,
+  title,
+  branding = true,
+  children,
+  className,
+  onPlay,
+  onPause,
+  onLoadedMetadata,
+  onPlaying,
+  onTimeUpdate,
+  ref,
+  ...rest
+}: VideoPlayerProps) => {
+  const videoPlayerRef = useRef<HTMLVideoElement>(null)
+  const videoWrapperRef = useRef<HTMLDivElement>(null)
+  const videoRef = ref ? ref : videoPlayerRef
+  const [playing, setPlaying] = useState(false)
+  const [playedTime, setPlayedTime] = useState(0)
+  const [totalTime, setTotalTime] = useState(0)
+  const [trackInformation, setTrackInformation] = useState<TextTrackCueList | undefined>(undefined)
+
+  /* > Video Attr Functions                                            */
```

Comment:
```suggestion
  /* Video Attr Functions  */
```

Reply:
Are we avoiding stylistic separators? Happy to update them all but just curious.